### PR TITLE
add Result.flatMap and tests

### PR DESCRIPTION
**Pull request by weissi**

### CODE REVIEW

File: Sources/Basic/Result.swift [hunk 1]
```diff
@@ -57,6 +57,15 @@ public enum Result<Value, ErrorType: Swift.Error> {
         }
     }
 
+    public func flatMap<U>(_ transform: (Value) -> Result<U, ErrorType>) -> Result<U, ErrorType> {
```
Comment (aciidb0mb3r) [inline Sources/Basic/Result.swift]: Doc comment pls :)

**Comment by aciidb0mb3r** [comment]
LGTM

File: Sources/Basic/Result.swift (hunk 1 above)
Comment (weissi) [inline Sources/Basic/Result.swift]: added one :)

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Pull request merged**